### PR TITLE
Rwc2017 challenge spr

### PR DESCRIPTION
**Pull request by Rayman**

### CODE REVIEW

File: challenge_spr/src/challenge_spr_states/bluff_game.py [hunk 1]
```diff
@@ -19,53 +20,65 @@
...
 
-    # Calculate params
-    th = desired_th - robot_th
-    if th > 3.1415:
-        th -= 2 * 3.1415
-    if th < -3.1415:
-        th += 2 * 3.1415
-    vth = 0.5
+    # Try to find a source for a couple of seconds
+    while not yaw and (rospy.Time.now() - start).to_sec() < 6:
```
Comment (MatthijsBurgh) [inline challenge_spr/src/challenge_spr_states/bluff_game.py]: maybe parametrize

File: challenge_spr/src/challenge_spr_states/bluff_game.py [hunk 2]
```diff
@@ -79,9 +92,12 @@ def execute(self, userdata):
 
         self.robot.head.look_at_standing_person()
 
+        t = threading.Thread(target=turn_to_closest_entity, args=(self.robot,))
```
Comment (MatthijsBurgh) [inline challenge_spr/src/challenge_spr_states/bluff_game.py]: Do we prefer threading over concurrence states?

File: challenge_spr/src/challenge_spr_states/riddle_game.py [hunk 3]
```diff
@@ -25,141 +26,162 @@ def hear(robot, time_out):
...
 
-                    if action['entity'] == 'boys':
-                        answer = 'In the crowd are %d boys' % crowd_data['boys']
+                        else:
+                            answer = 'I should name a category but I dont know %s' % entity
 
-                    if action['entity'] == 'girls':
-                        answer = 'In the crowd are %d girls' % crowd_data['girls']
+                    if action['action'] == "c_count":
+                        answer = "4 people are wearing %s" % str(action['entity'])
```
Comment (MatthijsBurgh) [inline challenge_spr/src/challenge_spr_states/riddle_game.py]: 4 has not a possibility in my opinion :p

File: challenge_spr/src/challenge_spr_states/bluff_game.py [hunk 4]
```diff
@@ -19,53 +20,65 @@
 
 from riddle_game import hear, answer
 
+# def _turn_to_closest_entity(robot):
```
Comment (jlunenburg) [inline challenge_spr/src/challenge_spr_states/bluff_game.py]: @SamAlexandrov Is this code deprecated? If so, can you remove it?

File: challenge_spr/src/challenge_spr_states/bluff_game.py [hunk 5]
```diff
@@ -19,53 +20,65 @@
...
-        print operator
-        if not operator:
-            vth = 0.5
-            th = 3.1415 / 10
-            print "Turning %f radians with force drive" % th
-            robot.base.force_drive(0, 0, vth, th / vth)
+def turn_to_closest_entity(robot):
 
-    robot.base.force_drive(0, 0, 0, 0.5)
+    rospy.sleep(1.5)
```
Comment (jlunenburg) [inline challenge_spr/src/challenge_spr_states/bluff_game.py]: @SamAlexandrov why do we need this sleep? Can we make it configurable (or at least define it at the top of this file)

File: challenge_spr/src/challenge_spr_states/bluff_game.py (hunk 1 above)
Comment (jlunenburg) [reply challenge_spr/src/challenge_spr_states/bluff_game.py]: @SamAlexandrov Also see my previous comment

File: challenge_spr/src/challenge_spr_states/riddle_game.py [hunk 6]
```diff
@@ -25,141 +26,162 @@ def hear(robot, time_out):
...
-                # Predefined questions
-                if action['action'] == 'answer':
-                    # if res.semantics['actions'] == 'answer':
-                    answer = action['solution']#res.semantics['solution']
+                        if action['entity'] == 'children':
+                            answer = 'In the crowd are %d children' % crowd_data['children']
 
-                # Counting people
-                if action['action'] == 'count':
+                        if action['entity'] == 'adults':
```
Comment (jlunenburg) [inline challenge_spr/src/challenge_spr_states/riddle_game.py]: @SamAlexandrov We might cook up a more elegant solution here. Not necessarily in this merge request though

**Comment by jlunenburg** [comment]
@SamAlexandrov Two changes requested:
- get rid of the magic numbers as mentioned in the comments
- add docstrings to your code. That would make it much easier for other people to re-use it.

File: challenge_spr/src/challenge_spr_states/riddle_game.py [hunk 7]
```diff
@@ -25,141 +26,162 @@ def hear(robot, time_out):
...
+                        if len(locations) == 1:
+                            loc = locations[0]['room']
+                            answer = 'The %s can be found in the %s' % (entity, loc)
+                        else:
+                            answer = 'I dont know that object'
+
+                    # Count placements or beacons in the room
+                    if action['action'] == 'count_placement':
+                        entity = action['entity']
+                        locations = [l for l in common_knowledge.locations if l['name'] == entity]
```
Comment (alberth) [inline challenge_spr/src/challenge_spr_states/riddle_game.py]: In general, avoid l, and o as variables, as they look a lot like 1 and 0 in some fonts. Better yet, use more than one letter (eg "loc" here)

File: challenge_spr/src/challenge_spr_states/riddle_game.py [hunk 8]
```diff
@@ -1,207 +1,215 @@
 #!/usr/bin/python
 
-import roslib;
+import roslib
```
Comment (jlunenburg) [inline challenge_spr/src/challenge_spr_states/riddle_game.py]: Do we actually need roslib? And the shebang?

Comment (LoyVanBeek) [reply challenge_spr/src/challenge_spr_states/riddle_game.py]: Not anymore, since ages, I think. Seems like a copy paste error for ye olde code

**Pull request merged**